Kitodo/src/main/java/org/kitodo/production/services/file/FileService.java

```diff
@@ -1281,4 +1284,25 @@ public URI deleteFirstSlashFromPath(URI uri) {
         }
         return URI.create(uriString);
     }
+
+    /**
+     * Check and return whether process with given ID has an empty generator folder or not.
+     *
+     * @param processId process ID
+     * @return whether given URI points to empty directory or not
+     * @throws IOException thrown if listing contents of given URI is not possible
+     */
+    public static boolean hasImages(int processId) throws IOException, DAOException {
+        Process process = ServiceManager.getProcessService().getById(processId);
+        URI processUri = ServiceManager.getProcessService().getProcessDataDirectory(process);
+        String sourceDir = processUri.toString() + process.getProject().getGeneratorSource().getRelativePath();
```

Comment:
This will fail if `generatorSource` is `null`.

Reply:
This won’t work. `VariableReplacer` needs to be invoked on the path returned for generatorSource (for example “images/(processtitle)_tif” must be converted to “images/1234567X_LoremIpsum_tif”)

Reply:
It did work in the examples I tested because I didn't use variables in the folder configuration of my project. Thanks for pointing this out.